test/kitchen/tasks/kitchen.py

```diff
@@ -171,16 +171,31 @@ def should_rerun_failed(_, runlog):
     """
     Parse a log from kitchen run and see if we should rerun it (e.g. because of a network issue).
     """
-    test_result_re = re.compile(r'\d+\s+examples?,\s+(?P<failures>\d+)\s+failures?')
+    test_result_re_gotest = re.compile(r'--- FAIL: (?P<failures>[A-Z].*) \(.*\)')
+    test_result_re_rspec = re.compile(r'\d+\s+examples?,\s+(?P<failures>\d+)\s+failures?')
+
     with open(runlog, 'r', encoding='utf-8') as f:
         text = f.read()
-        result = set(test_result_re.findall(text))
+        result_rspec = set(test_result_re_rspec.findall(text))
+        result_gotest = set(test_result_re_gotest.findall(text))
+        result = result_rspec.union(result_gotest)
         if result == {'0'} or result == set():
             print("Seeing no failed tests in log, advising to rerun")
         else:
             raise Exit("Seeing some failed tests in log, not advising to rerun", 1)
 
 
+@task
+def invoke_unit_tests(ctx):
+    """
+    Run the unit tests on the invoke tasks
+    """
+    for _, _, files in os.walk("tasks/unit-tests/"):
+        for file in files:
+            if file[-3:] == ".py" and file != "__init__.py":
+                ctx.run(f"python3 -m tasks.unit-tests.{file[:-3]}")
```

Comment:
I'm wondering if we could use
`python -m unittest discover -s tasks/unit-tests/ -p "*_test.py"`
instead

Reply:
Maybe we could do this, but I just copy pasted the invoke task we use in `tasks/tests.py`.

Reply:
`python3` won't work on windows, it only has `python`. 

output from the pipeline
```
You should consider upgrading via the 'C:\embeddedpy\py3.9.17-8236689\python.exe -m pip install --upgrade pip' command.
'python3' is not recognized as an internal or external command,

operable program or batch file.

[1;37mpython3 -m tasks.unit-tests.git_api_tests[0m
'python3' is not recognized as an internal or external command,

operable program or batch file.

[1;37mpython3 -m tasks.unit-tests.kitchen_unit_tests[0m
```
since the job still passed with those errors I think you'll need to check the return value from `ctx.run()`

[sys.executable](https://docs.python.org/3/library/sys.html#sys.executable) might be the best platform agnostic way to run with the same python version that is running the invoke task.

alternatively Windows also has `py.exe` and you can run specific versions of python with `py.exe -3 <command>`

Reply:
Thanks for your comment!
Indeed we did not realize the tests were not run on Windows. The behaviour is very strange, it looks like when the `ctx.run()` command in ran and it does not recognize python3, it automatically exit the invoke task (with an exit code 0 since it does not fail the pipeline), thus I cannot really  check the return value of `ctx.run()`
Still I changed the invoke task to use `sys.executable`

Reply:
I found out what the problem was, failure on the unit tests were not catch in the `.ps1` file running the test.
Now the test will fail if unit tests does not succeed.